src/components/add-journey/hooks.ts

```diff
@@ -0,0 +1,46 @@
+import axios from 'axios';
+import toast from 'react-hot-toast';
+import { TRANSPORT_API_URL } from '@/constants';
+import { useJourneySearchStore } from '@/hooks/useJourneySearchStore';
+import type { Journey, Station } from '@/types/opendata';
+import { useQuery } from '@tanstack/react-query';
+
+const getJourneys = ({
+  departureStation,
+  arrivalStation,
+  departureTime,
+}: {
+  readonly departureStation?: Station;
+  readonly arrivalStation?: Station;
+  readonly departureTime: string;
+}) => {
+  return axios.get<{ connections: Journey[] }>(
+    `${TRANSPORT_API_URL}/connections?from=${departureStation?.name}&to=${arrivalStation?.name}&date=${
+      departureTime.split('T')[0]
+    }&time=${departureTime.split('T')[1]}`
+  );
+};
+
+export const useGetJourneys = () => {
```

Comment:
Nice job with the hook! I like using `react-query for this`. But I'd move the file to `src/hooks/useGetJourneys.ts` since we keep all hooks centrally in this project.